src/api/utils/auth.py

```diff
@@ -115,24 +121,25 @@ def store_user_details(payload, token):
         Role.query.filter_by(name=role).first() for role in roles
         if role != "Andelan" and Role.query.filter_by(
             name=role).first() is not None]
-
     user.save()
-
-    g.current_user = user
-    g.current_user_token = token
+    return user
 
 
 def roles_required(roles):  # roles should be a list
     """Ensure only authorised roles may access sensitive data."""
     def check_user_role(f):
         @wraps(f)
         def decorated(*args, **kwargs):
-            if not set(g.current_user.roles).issubset(
-                    [Role.query.filter_by(name=role).first()
-                     for role in roles]):
-                return response_builder(dict(message="You're unauthorized"
-                                             " to perform this operation"),
-                                        401)
+            route_roles = list(filter(lambda r: r is not None,
+                                      [Role.query.filter_by(name=role).first()
+                                       for role in roles]))
+            for role in g.current_user.roles:
+                if role in route_roles:
+                    break
+            else:
+                return response_builder(
+                    dict(message="You're unauthorized"
+                         " to perform this operation"), 401)
```

Comment:
Could you walk me thorough how this works?

Reply:
@Andretalik Loops through required roles and user roles, if the user has any of the required roles the loop break and user is allowed to carry out the operation if not and the loop runs through the else block executes.